src/main/java/com/salesforce/dynamodbv2/mt/mappers/metadata/PrimaryKey.java

```diff
@@ -47,6 +48,8 @@ public PrimaryKey(String hashKey,
                       ScalarAttributeType hashKeyType,
                       Optional<String> rangeKey,
                       Optional<ScalarAttributeType> rangeKeyType) {
+        Preconditions.checkState(rangeKey.isPresent() == rangeKeyType.isPresent(),
```

Comment:
(FWIW, this was inspired by some get() calls on rangeKeyTypes without isPresent() checks--because they're in a scope where rangeKey.isPresent() is true)